geth/notification/fcm/notifier.go

```diff
@@ -0,0 +1,52 @@
+package fcm
+
+import (
+	"github.com/NaySoftware/go-fcm"
+	"github.com/status-im/status-go/geth/notification"
+)
+
+// Notifier represents messaging provider for notifications.
+type Notifier struct {
+	firebaseClient
+}
+
+// NewNotifier Firebase Cloud Messaging client constructor.
+func NewNotifier(key string) *Notifier {
+	return &Notifier{fcm.NewFcmClient(key)}
+}
+
+// Notify preparation and send to the tokens list.
+func (p *Notifier) Notify(body interface{}, tokens ...string) error {
+	p.setPayload(&notification.Payload{
+		Title: "Status - new message",
+		Body:  "ping",
+	})
+
+	p.setMessage(body, tokens...)
```

Comment:
As I can see, this object is mutable and `Notify` signal is possible to be invoked in parallel. Can you, please, write a test which shows there's no race conditions? From the first glance, here will be some.

Reply:
ok